File: client-clojure/src/sirqul_io_t_platform/specs/category_tree_response.clj
(ns sirqul-io-t-platform.specs.category-tree-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.category-tree-response :refer :all]
            )
  (:import (java.io File)))


(def category-tree-response-data
  {
   (ds/opt :categoryId) int?
   (ds/opt :name) string?
   (ds/opt :display) string?
   (ds/opt :subCategoryCount) int?
   (ds/opt :parentId) int?
   (ds/opt :parentName) string?
   (ds/opt :description) string?
   (ds/opt :active) boolean?
   (ds/opt :asset) asset-short-response-spec
   (ds/opt :favoriteId) int?
   (ds/opt :favorite) boolean?
   (ds/opt :applicationId) int?
   (ds/opt :type) string?
   (ds/opt :externalId) string?
   (ds/opt :externalType) string?
   (ds/opt :children) (s/coll-of category-tree-response-spec)
   })

(def category-tree-response-spec
  (ds/spec
    {:name ::category-tree-response
     :spec category-tree-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/CategoryTreeResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import CategoryTreeResponse._

case class CategoryTreeResponse (
  categoryId: Option[Long],
name: Option[String],
display: Option[String],
subCategoryCount: Option[Long],
parentId: Option[Long],
parentName: Option[String],
description: Option[String],
active: Option[Boolean],
asset: Option[AssetShortResponse],
favoriteId: Option[Long],
favorite: Option[Boolean],
applicationId: Option[Long],
`type`: Option[String],
externalId: Option[String],
externalType: Option[String],
children: Option[List[CategoryTreeResponse]])

object CategoryTreeResponse {
  import DateTimeCodecs._

  implicit val CategoryTreeResponseCodecJson: CodecJson[CategoryTreeResponse] = CodecJson.derive[CategoryTreeResponse]
  implicit val CategoryTreeResponseDecoder: EntityDecoder[CategoryTreeResponse] = jsonOf[CategoryTreeResponse]
  implicit val CategoryTreeResponseEncoder: EntityEncoder[CategoryTreeResponse] = jsonEncoderOf[CategoryTreeResponse]
}
